test/edu/washington/escience/myria/operator/StreamingAggTest.java

```diff
@@ -0,0 +1,1093 @@
+package edu.washington.escience.myria.operator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.joda.time.DateTime;
+import org.junit.Test;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.operator.agg.MultiGroupByAggregate;
+import edu.washington.escience.myria.operator.agg.PrimitiveAggregator.AggregationOp;
+import edu.washington.escience.myria.operator.agg.SingleColumnAggregatorFactory;
+import edu.washington.escience.myria.operator.agg.SingleGroupByAggregate;
+import edu.washington.escience.myria.operator.agg.StreamingAggregate;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleBatchBuffer;
+
+/**
+ * Test cases for {@link StreamingAggregate} class. Source tuples are generated in sorted order on group keys, if any.
+ * Some of the tests are taken from those for {@link SingleGroupByAggregate} and {@link MultiGroupByAggregate} since
+ * StreamingAggregate is expected to behave the same way they do if input is sorted.
+ */
+public class StreamingAggTest {
+
+  /**
+   * Construct a TupleBatchBuffer to be used as source of aggregate. Fixed schema and sorted on grouping columns.
+   *
+   * @param numTuples number of tuples to be added
+   * @return filled TupleBatchBuffer with each group key having (numTuples/10) tuples
+   */
+  private TupleBatchBuffer fillInputTbb(final int numTuples) {
+    final Schema schema =
+        Schema.ofFields(Type.INT_TYPE, "Int", Type.DOUBLE_TYPE, "Double", Type.FLOAT_TYPE, "Float", Type.LONG_TYPE,
+            "Long", Type.DATETIME_TYPE, "Datetime", Type.STRING_TYPE, "String", Type.BOOLEAN_TYPE, "Boolean",
+            Type.LONG_TYPE, "value");
+
+    final TupleBatchBuffer source = new TupleBatchBuffer(schema);
+    for (int i = 0; i < numTuples; i++) {
+      int value = i / (numTuples / 10);
+      source.putInt(0, value);
+      source.putDouble(1, value);
+      source.putFloat(2, value);
+      source.putLong(3, value);
+      source.putDateTime(4, new DateTime(2010 + value, 1, 1, 0, 0));
+      source.putString(5, "" + value);
+      source.putBoolean(6, (i / (numTuples / 2) == 0));
+      source.putLong(7, 2L);
+    }
+    return source;
+  }
+
```

Comment:
As I hinted at earlier, I think you need at least one new unit test to verify the streaming semantics: one which generates more than one `TupleBatch` worth of aggregation results, and one (possibly the same test) which verifies that when multiple `TupleBatch`es of results are generated, the first `TupleBatch` of results is returned before the child is exhausted (i.e., the operator is actually "streaming").